### PR TITLE
cmake enabling

### LINKED ISSUE #106: [1.8.4] Cannot be built on Windows
I'm getting errors while trying to build `release-1.8.4` with `Visual Studio 2017`:
```
Cannot open include file: 'autoconfig.h': No such file or directory
Cannot open source file: '..\..\threadutil\src\FreeList.c': No such file or directory
Cannot open source file: '..\..\threadutil\src\LinkedList.c': No such file or directory
Cannot open source file: '..\..\threadutil\src\ThreadPool.c': No such file or directory
Cannot open source file: '..\..\threadutil\src\TimerThread.c': No such file or directory
Cannot open include file: 'upnpconfig.h': No such file or directory
```

### PR DESCRIPTION
**Pull request by Vollstrecker**
Hi,

here is the first version.

What's done: Building everything with windows and linux (No other working system in VMs here atm.).

pthreads4w will be found systemwide or in the choosen installation dir, if it's not found, and the option is set, pthreads4w will be downloaded from my forked repo and built and installed all in one with upnp itself.

What's missing:
Docs
Samples
Tests
Doxygen

It's only selectable to build debug or release on a global base. x86 or x64 is depending on the system where it is built. I didn't find anything dependent on that, but if it is in the detected values, it won't work as easy as expected.

The targets (tested with VS2019 only) are named when building release flavour:
- upnp (the usual lib)
- upnps (the static lib)
- ixml (the usual lib)
- ixmls (the static lib)

When building debug flavour, ervery lib will be postfixed by a "d", so it's possible to have them all install in parallel.

When INSTALL target is built, all 4 libs are built and installed. I didn't see any benefit in installing them separately.

[Edit]
Forgot the mention:
fixes #106

### CODE REVIEW

**Comment by Vollstrecker** [comment]
And yes, there are a lot of commits yesterday. That was the date when I rebased to current master.

**Comment by mrjimenez** [comment]
Great work, thanks Vollstrecker!

Regards,
Marcelo.

**Comment by Stullemon** [comment]
Thanks for this! This is much appreciated!

File: CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,511 @@
...
+INSTALL (FILES
+	${CMAKE_CURRENT_BINARY_DIR}/UPNP.cmake
+	${CMAKE_CURRENT_BINARY_DIR}/UPNPVersion.cmake
+	DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/UPNP
+)
+
+SET (VERSION ${UPNP_VERSION_STRING})
+SET (prefix ${CMAKE_INSTALL_PREFIX})
+SET (exec_prefix "\${prefix}")
+SET (libdir "\${exec_prefix}/${CMAKE_INSTALL_LIBDIR}")
```
Comment (jtojnar) [inline CMakeLists.txt:499]: Note that `CMAKE_INSTALL_*` variables might be absolute paths, in which case the pkg-config will contain wrong paths. See also https://github.com/jtojnar/cmake-snips#concatenating-paths-when-building-pkg-config-files

Comment (Vollstrecker) [reply CMakeLists.txt:499]: Good to know. I'll take a look at this in the next week.

**Pull request merged**